docs/adr/0010-transition-fxa-from-backbone-to-react.md

```diff
@@ -0,0 +1,151 @@
+
+Note: also see pros/cons listed on Backbone in general [in this ADR](https://github.com/mozilla/fxa/blob/master/docs/adr/0002-use-react-redux-and-typescript-for-subscription-management-pages.md)
+
+### Options B - E - Common Pros & Cons
+
+These pros and cons are shared for the rest of the choices. The pros and cons listed for all subsequent options will be in comparison to _each other_ and not against option A.
+
+- Description
+  - Move away from Backbone/Mustache in the content server in favor of another option.
+- Pros
+  - Future-proof the FE stack for the foreseeable future: modern libraries and frameworks have significant backings and resources placed to work on them (i.e. Angular by Google, React by Facebook) and are continuing to gain community and feature support.
+  - Benefit from the modern architectural approaches and features other technologies offer.
+  - Increase the chance of new engineers to the team as well as open source contributors being familiar with the stack.
+  - While refactoring, our tests can be assessed for need and clarity. A fresh pair of eyes on these can point out superfluous tests or those we lack.
+- Neutral
+  - Modern libraries and frameworks (React, Angular, VueJS) all offer some flavor of virtual DOM for rerenders and DOM manipulations, and offer performance boosts and optimizations out of the box that Backbone does not offer. To varying degree, we would expect to see a performance boost, but there may be cases in which a granular DOM update would be faster without a virtual DOM.
+- Cons
+  - We will gain tech debt regarding a large refactor effort that may be ongoing for some time.
+  - A lot of work was put into FxA and moving away from Backbone won't be a simple task. There is a lot of infrastructure we must ensure support for and respect the integrity of.
+  - A transition from Backbone to any of the options won't be 1:1 - that is to say, Backbone and Mustache closely resemble the common "MVC" pattern, where Backbone models tied to collections represent the "M", Backbone views act as the "C", and our Mustache templates act as the "V" layer. Angular and Vue appear to follow patterns more closely described as "MVVM", and while React can broadly be seen as the "V" in "MVC", React architecture typically follows the Flux design pattern. We will need to carefully translate pieces of Backbone models and controllers into state, helper functions, or Typescript interfaces where applicable.
+  - Whatever new library or framework we choose will require a learning curve for at least part of the team. It may slow the ability to make front-end changes at first.
+
+### Option B - Move to client-side React
+
+- Description:
+  - Use client-side React.
+- Pros
+  - React is backed by a well-known company (Facebook) with a lot of dedicated resources working on the library and it's popular, has a lot of community support, and has ample documentation.
```

Comment:
are there any concerns when/as Facebook's popularity decreases?

Reply:
@clouserw That's a good consideration. I think React's current very high level of popularity can help mitigate some risk around that, but it's hard to predict what will happen (-cough- MySpace). I'll add a comment around this somewhere.

@lmorchard I actually have web components in the "Other considerations" option. Are you talking about using web components with a web component library like [stencilJS](https://stenciljs.com/)? I'm having a hard time picturing how we could get there otherwise, they seem too... simplistic?

Reply:
More something like Web Components plus [lit-html](https://lit-html.polymer-project.org/guide) / LitElement, which lets you do functional templates like React but doesn't use a virtual DOM. I don't think we should adopt it now, but it could be worth watching